### PR TITLE
Buy and Sell page

### PR DESCRIPTION
**Pull request by ParadoxicalNerd**

### CODE REVIEW

File: client/src/Pages/BuyAndSellPage/index.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    const [companyDetails, setCompanyDetails] = useState()
+
+    const [value, setValue] = useState(0);
+
+    /**
+     * 
+     * @param {int} value Number of stocks in transaction
+     * @param {boolean} buy Is it a buy or a sell transaction
+     */
+    const onSubmitHandler = async (value, buy) => {
```
Comment (MaxDillon) [inline client/src/Pages/BuyAndSellPage/index.js:27]: Looks great. Might want to consider putting this func outside the functional component. Just cuz javascript creates a new func instance on every draw. It's not really a problem, but also a really easy fix

File: client/src/Pages/BuyAndSellPage/index.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+/*
+TODO:
+[] Figure out cors
+[] 
+*/
+
+
+export default () => {
+
+    const companyID = "uber"
```
Comment (MaxDillon) [inline client/src/Pages/BuyAndSellPage/index.js:14]: One thing you can do is use the react router handlers. In the router, you can say 

\<Route path="trade/:company_name" element={\<BuyAndSellPage /\>} /\>

Where conpany_name is the company you want to buy or sell (the colon specifies it's an arguent). You can then use the useParams() hook in this function to get the company id/name parameter and do what you will with it

File: client/src/Pages/BuyAndSellPage/index.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            })
+
+            const prices = parsedResponse.data.stockData.map(row => row.trading_price)
+
+            // console.log(date_data)
+            setData(() => ([{
+                x: dates,
+                y: prices,
+                type: 'scatter'
+            }]))
```
Comment (MaxDillon) [inline client/src/Pages/BuyAndSellPage/index.js:68]: Might wanna consider extrating this to a function for cleanliness sake

File: server/blueprints/buySellBlueprint.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    @buySellBlueprint.route('/', methods=['GET'])
+    @login_required(Makesession)
+    def getStockDetails():
+
+        response_body = {
+            "error": None,
+            "data": None
+        }
+
+        session: Session = Makesession()
```
Comment (MaxDillon) [inline server/blueprints/buySellBlueprint.py:23]: Maybe create another python package and file and extract some of this functions functionality. It looks great. I'm more thinking from the perpsective of organization

File: server/blueprints/buySellBlueprint.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+    @buySellBlueprint.route('/', methods=['POST'])
+    @login_required(Makesession)
+    def executeTransaction():
+        response_body = {
+            "error": None,
+            "ok": False
+        }
+
+        session: Session = Makesession()
```
Comment (MaxDillon) [inline server/blueprints/buySellBlueprint.py:65]: Same here, maybe extract to another function

**Review by MaxDillon — COMMENTED** [review]
This all looks really good. I know you didn't request a review yet. I don't want to rush you, but apparently most PR's are dependent on yours rn so...

Everything I say is mostly a nitpick or about style/cleanliness. I think this will be in good shape for merge after getting optipn for passing different company id's

File: server/auth/auth_tools.py [hunk 6]
```diff
@@ -6,53 +6,51 @@
...
 def get_accounts(session: Session):
-	account_query = select(Account.user_id, Account.username, cast(Account.pass_hash, String).label("pass_hash"))
-	accounts = session.execute(account_query).all()
-
-	return {'data':[dict(account) for account in accounts]}
+    account_query = select(user_id, username, cast(
+        pass_hash, String).label("pass_hash"))
+    accounts = session.execute(account_query).all()
 
+    return {'data': [dict(account) for account in accounts]}
```
[the remark this answers is not in the document]
Comment (ParadoxicalNerd) [reply server/auth/auth_tools.py:14]: That's weird, I don't remember touching it at all. I prefixed the stuff with Account

File: server/requirements.txt [hunk 7]
```diff
@@ -9,4 +9,5 @@ werkzeug==2.0.2
 jinja2<3.1.0
 numpy
 pandas
-yfinance
\ No newline at end of file
+yfinance
+flask-cors
```
[the remark this answers is not in the document]
Comment (ParadoxicalNerd) [reply server/requirements.txt:13]: No

File: client/src/Pages/BuyAndSellPage/index.js (hunk 1 above)
Comment (ParadoxicalNerd) [reply client/src/Pages/BuyAndSellPage/index.js:27]: Done. I did not realize this

File: client/src/Pages/BuyAndSellPage/index.js (hunk 3 above)
Comment (ParadoxicalNerd) [reply client/src/Pages/BuyAndSellPage/index.js:68]: Do you want me to extract the dates bit or dates and prices?

File: server/blueprints/buySellBlueprint.py (hunk 5 above)
Comment (ParadoxicalNerd) [reply server/blueprints/buySellBlueprint.py:65]: Where do you think I should put the function?

File: client/src/Pages/BuyAndSellPage/index.js (hunk 3 above)
Comment (MaxDillon) [reply client/src/Pages/BuyAndSellPage/index.js:68]: I was thinking both probably, but totally up to you how you do it. Maybe both in 1 func. Maybe both in seperate funcs. I've just generally been told to keep what's in the body of the functional component to only what's necesarry e.g. hooks and return statements mostly

File: server/blueprints/buySellBlueprint.py (hunk 5 above)
Comment (MaxDillon) [reply server/blueprints/buySellBlueprint.py:65]: That was a bit confusing I realize.The function it'self is good, I just noticed that the body of the func was a bit long and could be broken up into a package for some type of functionality

Comment (MaxDillon) [reply server/blueprints/buySellBlueprint.py:65]: That way you can read over the blueprint and it's really clear what it takes as input, what it does, and what it gives as output

**Review by MaxDillon — APPROVED** [review]
Looks great! You can merge whenever you think it's ready. At some point, we probably want to transfer your functions to a py file outside of blueprint, but that doesn't need to be in this PR. 

Also, there is a bit of a strange inerraction between the page and the navbar overlapping. Not a big deal yet, but probably a problam to solve at some point.

Also, the way you have a get and a post to "/getAccounts/" works fine, but cuz they have different functionalities, it would probably be best for them each to have their own unique path. 

If you do make some of these changes before merging, you don't need to re request review, and if you don't, that's fine. We can handle them in future PR's

**Pull request merged**